src/main/java/com/fortanix/sdkms/performance/helper/AESHelper.java

```diff
@@ -51,19 +51,21 @@ public EncryptRequestEx createEncryptRequest(SobjectDescriptor key, String plain
     }
 
     @Override
-    public DecryptRequestEx createDecryptRequest(SobjectDescriptor key, byte[] cipher) {
+    public DecryptRequestEx createDecryptRequest(SobjectDescriptor key, byte[] cipher, byte[] tag) {
         // generate tag
-        byte[] tag = null;
-        if (mode == CryptMode.GCM || mode == CryptMode.CCM) {
-            int tagLengthInByte = AES_BLOCK_SIZE;
-            tag = new byte[tagLengthInByte];
-            byte[] temp = new byte[cipher.length - tagLengthInByte];
-            // separate out GCM tag
-            System.arraycopy(cipher, 0, tag, 0, tagLengthInByte);
-            // separate out cipher text
-            System.arraycopy(cipher, tagLengthInByte, temp, 0, temp.length);
-            cipher = temp;
-        }
+//        byte[] tag = null;
+//        if (mode == CryptMode.GCM || mode == CryptMode.CCM) {
+//            int tagLengthInByte = AES_BLOCK_SIZE;
+//            tag = new byte[tagLengthInByte];
+//            byte[] temp = new byte[cipher.length - tagLengthInByte];
+//            // separate out GCM tag
+//            System.arraycopy(cipher, 0, tag, 0, tagLengthInByte);
+//            // separate out cipher text
+//            System.arraycopy(cipher, tagLengthInByte, temp, 0, temp.length);
+//            cipher = temp;
+//        }
+
+        // Getting the Tag value from the Encrypt request
         DecryptRequestEx decryptRequest = new DecryptRequestEx().key(key).alg(this.algorithm).cipher(cipher).mode(this.mode).ad(this.ad).tag(tag);
```

Comment:
AESHelper extends EncryptionDecryptionHelper. Hence we can directly call super.decryptRequest(...).
Infact I would suggest to update the decryptRequest method in EncryptionDecryptionHelper to take IV as the input, so that super.decryptRequest(...) can be called in line 71 too. (See my other comment in EncryptionDecryptionHelper.java)